src/gas.rs

```diff
@@ -8,10 +8,20 @@
 // instructions can operate on them efficiently.
 
 use core::ops::Range;
+use std::cmp::{max, min};
 
 /// Type alias for gas
 pub type Gas = u64;
 
+/// Constants for inter-contract call gas reserve calculations
```

Comment:
The use of floating point math here could introduce different rounding in subsequent calls to gas_left etc, we should be using fixed-point math for these calculations.